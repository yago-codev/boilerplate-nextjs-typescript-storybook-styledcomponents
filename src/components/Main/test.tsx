import { screen, render } from '@testing-library/react';

import Main from './index';

describe('</Main>', () => {
  it('Deve renderizar um título dentro de uma tag <h>', () => {
    const { container } = render(<Main />);

    expect(
      screen.getByRole('heading', { name: /react avançado/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('Deve renderizar as cores corretamente', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toHaveStyle({
      'background-color': '#06092b',
    });
  });
});
