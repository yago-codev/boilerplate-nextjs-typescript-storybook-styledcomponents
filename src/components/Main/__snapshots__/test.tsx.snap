// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</Main> Deve renderizar um título dentro de uma tag <h> 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  background-color: #06092b;
  color: #fff;
  padding: 3rem;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  width: min(30rem,100%);
  margin-top: 3rem;
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado."
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado
  </h1>
  <h2
    class="c3"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com código."
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`;
